internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -0,0 +1,233 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+
+	"github.com/rancher/fleet/internal/cmd/controller/summary"
+	v1alpha1 "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/fleet/pkg/durations"
+	"github.com/rancher/fleet/pkg/sharding"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type StatusReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	Workers int
+	ShardID string
+}
+
+func (r *StatusReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.GitRepo{}).
+		Watches(
+			// Fan out from bundle to gitrepo
+			&v1alpha1.Bundle{},
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, a client.Object) []ctrl.Request {
+				repo := a.GetLabels()[v1alpha1.RepoLabel]
+				if repo != "" {
+					return []ctrl.Request{{
+						NamespacedName: types.NamespacedName{
+							Namespace: a.GetNamespace(),
+							Name:      repo,
+						},
+					}}
+				}
+
+				return []ctrl.Request{}
+			}),
+			builder.WithPredicates(bundleStatusChangedPredicate()),
+		).
+		WithEventFilter(sharding.FilterByShardID(r.ShardID)).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.Workers}).
+		Named("GitRepoStatus").
+		Complete(r)
+}
+
+// Reconcile reads the stat of the GitRepo and BundleDeployments and
+// computes status fields for the GitRepo. This status is used to
+// display information to the user.
+func (r *StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithName("gitops-status")
+	gitrepo := &v1alpha1.GitRepo{}
+
+	if err := r.Get(ctx, req.NamespacedName, gitrepo); err != nil && !errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	} else if errors.IsNotFound(err) {
+		logger.V(1).Info("Gitrepo deleted, cleaning up poll jobs")
```

Comment:
nit: perhaps this log message should not be produced from here, but rather from the gitops reconciler.